includes/query_processor.inc

```diff
@@ -232,6 +236,9 @@ class IslandoraSolrQueryProcessor {
       $params_date_facets["facet.date.start"] = 'NOW/YEAR-20YEARS';
```

Comment:
Question here: seems like facet.range was introduced in Solr 3.3+, so is there really a need to maintain deprecated on solr 6.6 facet.date piece? can we just get rid of it?

Reply:
I don't know, I just assume that someone somewhere is relying on a very old piece.

Reply:
Ok, cool we can discuss this on our next committers call. Won't block this pull but good to have consensus. Probably more needed to be really Solr 7 compliant in the future

Reply:
Almost... wondering if we should instead move to a factory, to facilitate the use of different implementations for different versions of Solr... the [version of Solr can be requested of Solr relatively easily](https://stackoverflow.com/a/48835496), hitting:
`http://whatever:port/solr/admin/info/system?wt=json`, giving something like (added in `indent=true` to the query parameters for formatting):
```json
{
  "responseHeader":{
    "status":0,
    "QTime":5},
  "mode":"std",
  "lucene":{
    "solr-spec-version":"4.10.4",
    "solr-impl-version":"4.10.4 1662817 - mike - 2015-02-27 16:43:07",
    "lucene-spec-version":"4.10.4",
    "lucene-impl-version":"4.10.4 1662817 - mike - 2015-02-27 16:38:43"},
[... a bunch of environment stuff ...]}
```

Would we then want to make `IslandoraSolrQueryProcessor` class just proxy out to whatever target implementation?

... Not sure... maybe going too far with abstraction fun, especially given this module supposedly just "supports" Solr 4.2+.